azure/templates/post-deploy.yml

```diff
@@ -68,6 +68,35 @@ steps:
     workingDirectory: "$(Pipeline.Workspace)/s/$(SERVICE_NAME)/$(SERVICE_ARTIFACT_NAME)"
     retryCountOnTaskFailure: 2
 
+  - bash: |
+      set -ex
+      if ! [[ $APIGEE_ENVIRONMENT =~ .*-*sandbox ]]; then
+        counter=0
+        base_path="$SERVICE_BASE_PATH"
+        apigee_env="$ENVIRONMENT"
+
+        endpoint="https://${apigee_env}.api.service.nhs.uk/${base_path}/_status"
+        while [[ $counter -lt 11 ]]; do
+          response=$(curl -H "apikey: $(status-endpoint-api-key)" -s "$endpoint")
+          response_code=$(jq -r '.checks.healthcheck.responseCode' <<< "$response")
+          response_body=$(jq -r '.checks.healthcheck.outcome' <<< "$response")
+          if [ "$response_code" -eq 200 ] && [ "$response_body" == "OK" ]; then
+              echo "Status test successful"
+              break
+          else
+              echo "Waiting for $endpoint to return a 200 response with 'OK' body..."
+              ((counter=counter+1)) # Increment counter by 1
+              echo "Attempt $counter"
+              sleep 100
```

Comment:
How long does it usually take to come up? 100 second gaps seems extreme. How about 20 seconds and 30 attempts? That also makes for a nice round 10 mins.